### PR TITLE
feat(apex): add auto-terminate setting for orphaned Apex Language Server processes - W-23251304

### PR DESCRIPTION
**Pull request by daphne-sfdc**
## Summary

- Adds a new boolean VS Code setting `salesforcedx-vscode-apex.autoTerminateOrphanedProcesses` (default `false`) that silently kills orphaned Apex Language Server processes on activation when enabled.
- When the setting is `false`, extends the existing orphan-detection toast with an "Always Auto-Terminate" button that triggers a confirmation modal and persists the setting on user confirmation.
- All setting reads/writes are handled inline via the existing `SettingsService` from `salesforcedx-vscode-services`; no new service layer needed.
- Adds a retry mechanism for orphan detection such that during reloads of the VSCode window, the Apex LSP from the previous session that has its shutdown in-progress does not get caught by the orphan detection of the next session's activation, causing a false positive orphan.

## Plan

See [.claude/plans/W-23251304.md](.claude/plans/W-23251304.md)

## Reviewer notes

**Medium findings (not auto-fixed — design decisions required):**

- **Finding 1 (unit-test-critic):** The test mocks 3 services via hand-rolled stubs. Consider extracting `alwaysAutoTerminateConfirmation`'s post-confirm logic into a pure function taking a settings-writer callback to reduce stub burden. This is an architectural refactoring suggestion that requires a design decision about whether to break from the `Effect.fn` pattern used throughout the file. File: `packages/salesforcedx-vscode-apex/src/languageServerOrphanHandler.ts`

- **Finding 3 (thermo):** While the second `ExtensionProviderService` resolution at line 148 is now defensively guarded, the deeper structural suggestion (hoisting the `api` reference from `findOrphanedProcesses` return value so `checkAndResolveOrphanedLanguageServers` can reuse it) requires changing the return type of `findOrphanedProcesses` from `ProcessDetail[]` to `{ api, processes }` — a larger refactor spanning multiple call sites. File: `packages/salesforcedx-vscode-apex/src/languageServerOrphanHandler.ts`

- **Finding 5 (thermo):** `alwaysAutoTerminateConfirmation` still independently resolves `ExtensionProviderService` rather than receiving the settings service as a parameter from `promptOnce`. Passing it as a parameter would require changing the function signature from `Effect.fn()(function*())` to accept an argument, and updating all callers in the `promptOnce`/`getResolutionForOrphanProcesses` chain. This is idiomatic in Effect-TS (services are resolved via DI rather than parameter passing), so changing it requires a design decision. File: `packages/salesforcedx-vscode-apex/src/languageServerOrphanHandler.ts`

### What issues does this PR fix or reference?

@W-23251304@

### Functionality Before

When orphaned Apex Language Server processes are detected on activation, only a toast with "Terminate them?" / "Show processes" / "Dismiss" was shown. No way to suppress prompts permanently.

### Functionality After

- New setting `salesforcedx-vscode-apex.autoTerminateOrphanedProcesses` in Settings UI under "Salesforce Apex Configuration".
- When `true`: orphans are silently killed on activation without any prompt.
- When `false` (default): existing toast appears with an added "Always Auto-Terminate" button. Clicking it opens a modal confirming consequences (enables setting, kills now, can disable later via Settings). Confirm = enable + kill; Cancel = no-op.

---

🤖 Generated by auto-build pipeline. Original WI: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00002dZHkyYAG/view

### CODE REVIEW

**Review by mshanemc — DISMISSED** [review]
✓ auto-build-wi peer-approve: pre-cleared via /ai-auto approve comment

**Comment by daphne-sfdc** [comment]
I wasn't able to get the orphans popup with the latest VSIX in this branch after I implemented the retry process to check for orphaned Apex LSP 3 times before displaying the popup.  It's proof the retry process is working as intended. 🙂 I tested the below using an older version that had the auto-terminate setting implemented without the retry process, where I could consistently get the popup and check the functionality.

Get into a situation where there is 1 orphaned Apex language server.
Screenshot 2026-07-30 at 1 30 14 PM

- Click "Show Processes" -> orphans listed in Output Tab under dropdown "Apex" ✅
Screenshot 2026-07-30 at 1 30 29 PM<br>

- Click "Terminate Processes" -> click "Yes" -> orphan gets killed ✅
- Click "Terminate Processes" -> click "Cancel" -> orphan remains ✅
Screenshot 2026-07-30 at 1 30 39 PM<br>

- Click "Always Auto-Terminate" -> click "Confirm" -> orphan gets killed, **Salesforcedx-vscode-apex: Auto Terminate Orphaned Processes** becomes checked, reloading the VSCode window no longer produces the warning notifications because future orphans are automatically killed ✅
- Click "Always Auto-Terminate" -> click "Cancel" -> orphan remains, **Salesforcedx-vscode-apex: Auto Terminate Orphaned Processes** remains unchecked ✅
Screenshot 2026-07-30 at 1 31 07 PM
Screenshot 2026-07-30 at 1 31 27 PM

File: packages/salesforcedx-vscode-apex/src/languageServerOrphanHandler.ts [hunk 1]
```diff
@@ -131,23 +131,56 @@ const killOne = Effect.fn('apex.orphan.killOne')(function* (processInfo: Process
...
+  // (e.g. a previous session's LSP completing its own graceful shutdown).
+  let confirmedOrphans: ProcessDetail[] = [];
+  for (let i = 1; i <= numTries; i++) {
+    confirmedOrphans = yield* findOrphanedProcessesSafe();
+    if (confirmedOrphans.length === 0) {
+      yield* annotateRootSpan('orphanCount', 0);
+      return;
+    }
+  }
+  yield* annotateRootSpan('orphanCount', confirmedOrphans.length);
```
Comment (daphne-sfdc) [inline packages/salesforcedx-vscode-apex/src/languageServerOrphanHandler.ts]: As discussed during standup.  On fast laptops, we want to leave enough time during reloads such that the Apex LSP process from the previous session isn't caught by the orphan detector from the next session while it's in the shutdown process, creating a false positive orphan.

**Review by peternhale — APPROVED** [review]
## ✅ Approved

Excellent implementation with strong Effect-TS idioms and comprehensive test coverage. The retry mechanism elegantly handles the reload race condition, and the graceful degradation ensures no breaking changes.

**Strengths:**
- Robust retry loop (3 attempts with 2s delay) prevents false positives during VS Code reloads
- Proper error handling with graceful degradation on settings read/write failures
- Comprehensive test coverage including edge cases (modal cancellation, settings failures, orphan disappearing on retry)
- Well-internationalized with English and Japanese translations
- Zero-latency when auto-terminate is enabled (silent kill, no prompt)

**Next steps:**
Please capture the three unaddressed findings mentioned in the PR description (unit-test-critic and thermo) as follow-up user stories:
- Finding 1: Extract `alwaysAutoTerminateConfirmation` post-confirm logic into pure function
- Finding 3: Hoist `api` reference from `findOrphanedProcesses` return value
- Finding 5: Consider passing settings service as parameter vs. independent resolution

These are reasonable architectural refactorings to track as tech debt rather than blocking this PR.

**Manual verification checklist:**
- [ ] Setting appears in VS Code Settings UI under "Salesforce Apex Configuration"
- [ ] Confirmation modal flow works (button → modal → setting persisted)
- [ ] Auto-terminate works silently when enabled
